SignalServiceKit/src/Storage/OWSSessionStorage.m

```diff
@@ -177,6 +178,41 @@ + (YapDatabaseConnection *)dbConnection
     return OWSSessionStorage.sharedManager.dbConnection;
 }
 
+#pragma mark - Migration
+
+- (void)migrateCollection:(NSString *)collection fromStorage:(OWSStorage *)storage valueClass:(Class)valueClass
+{
+    OWSAssert(collection.length > 0);
+    OWSAssert(storage);
+
+    DDLogInfo(@"%@: migrating %@", self.logTag, collection);
+
+    NSMutableDictionary<NSString *, id> *collectionContents = [NSMutableDictionary new];
+
+    // 1. Read from old storage.
+    [storage.newDatabaseConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+        [transaction
+            enumerateKeysAndObjectsInCollection:collection
+                                     usingBlock:^(NSString *_Nonnull key, id _Nonnull value, BOOL *_Nonnull stop) {
+                                         if (![value isKindOfClass:valueClass]) {
+                                             OWSFail(
+                                                 @"Unexpected type: %@ in collection: %@.", [value class], collection);
+                                             return;
+                                         }
+
+                                         collectionContents[key] = value;
+                                     }];
+    }];
+
+    // 2. Write to new storage.
+    [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+        [collectionContents enumerateKeysAndObjectsUsingBlock:^(NSString *_Nonnull key, id value, BOOL *_Nonnull stop){
+        }];
```

Comment:
There's a missing line here - the bug is fixed in the next PR.

Reply:
Right.  There's a bunch more work though that I around improving this method and I moved it elsewhere, etc. hence why I don't want to just fix it here.